README.md

```diff
@@ -92,6 +95,15 @@ Returns true if file.contents is a Stream.
 ### isNull()
 Returns true if file.contents is null.
 
+### isDirectory()
+Returns true if file is a directory. File is considered a directory when:
+
+- `file.isNull()` is `true`
+- `file.stat` is an object
+- `file.stat.isDirectory()` returns `true`
+
+When marking file as a directory, do it by mocking the `fs.Stats` object, and passing it via `options.stat` property into the constructor. Vinyl needs to know the file is a directory from the get go, or it might result in an inconsistent path history (some entries ending in separator, some not).
```

Comment:
I don't like "When marking file as a directory".  Maybe something like "When constructing a Vinyl object, pass in a valid `fs.Stats` object. Vinyl needs to know the file is a directory from the get go, or it might result in an inconsistent path history (some entries ending in separator, some not).  If you are mocking the `fs.Stats` object, ensure it has the `isDirectory` method.", or something similar.